cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinder.java

```diff
@@ -0,0 +1,246 @@
+      for (Map.Entry<BrokerEntity, Double> entry  : currentValue.entrySet()) {
+        if (currentValue.get(entry.getKey()) > base * PEER_METRIC_MARGIN) {
+          detectedSlowBrokers.add(entry.getKey());
+        }
+      }
+    }
+  }
+
+  private SlowBrokers createSlowBrokersAnomaly(Map<BrokerEntity, Long> detectedBrokers,
+                                               boolean fixable,
+                                               boolean removeSlowBroker,
+                                               String description) {
+    Map<String, Object> parameterConfigOverrides = new HashMap<>(5);
+    parameterConfigOverrides.put(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG, _kafkaCruiseControl);
+    parameterConfigOverrides.put(METRIC_ANOMALY_DESCRIPTION_OBJECT_CONFIG, description);
+    parameterConfigOverrides.put(METRIC_ANOMALY_BROKER_ENTITIES_OBJECT_CONFIG, detectedBrokers);
+    parameterConfigOverrides.put(REMOVE_SLOW_BROKERS_CONFIG, removeSlowBroker);
+    parameterConfigOverrides.put(ANOMALY_DETECTION_TIME_MS_OBJECT_CONFIG, _kafkaCruiseControl.timeMs());
+    parameterConfigOverrides.put(METRIC_ANOMALY_FIXABLE_OBJECT_CONFIG, fixable);
+    return _kafkaCruiseControl.config().getConfiguredInstance(KafkaCruiseControlConfig.METRIC_ANOMALY_CLASS_CONFIG,
+                                                              SlowBrokers.class,
+                                                              parameterConfigOverrides);
+  }
+
+  private String getSlowBrokerDescription(Map<BrokerEntity, Long> detectedBrokers) {
+    StringBuilder descriptionSb = new StringBuilder().append("{\n");
+    detectedBrokers.forEach((key, value) -> {
+      descriptionSb.append("\tBroker ").append(key.brokerId()).append("'s performance degraded at ").append(toDateString(value)).append("\n");
+    });
+    descriptionSb.append("}");
+    return descriptionSb.toString();
+  }
+
+  @Override
+  public Collection<MetricAnomaly<BrokerEntity>> metricAnomalies(Map<BrokerEntity, ValuesAndExtrapolations> metricsHistoryByBroker,
+                                                                 Map<BrokerEntity, ValuesAndExtrapolations> currentMetricsByBroker) {
+    Set<MetricAnomaly<BrokerEntity>> detectedAnomalies = new HashSet<>();
+    try {
+      Set<BrokerEntity> detectedSlowBrokers = detectSlowBrokers(metricsHistoryByBroker, currentMetricsByBroker);
+
+      Map<BrokerEntity, Long> brokersToDemote = new HashMap<>();
+      Map<BrokerEntity, Long> brokersToRemove = new HashMap<>();
+      for (BrokerEntity broker : detectedSlowBrokers) {
+        // Update slow broker detection time and slowness score.
+        Long currentTimeMs = _kafkaCruiseControl.timeMs();
+        _detectedSlowBrokers.putIfAbsent(broker, currentTimeMs);
+        Integer slownessScore = _brokerSlownessScore.compute(broker,
+                                                             (k, v) -> (v == null) ? 1 : Math.min(v + 1, SLOW_BROKER_DECOMMISSION_SCORE));
```

Comment:
This slowness score is a little confusing. Can we split the content of `metricAnomalies` function to parseable smaller functions?